tests/StackExchange.Redis.Tests/SortedSets.cs

```diff
@@ -146,5 +176,468 @@ public async Task SortedSetPopMulti_Zero_Async()
+            }
+        }
+
+        [Fact]
+        public void SortedSetRangeStoreByLexExclusiveRange()
+        {
+            using var conn = Create();
+            Skip.IfMissingFeature(conn, nameof(RedisFeatures.SortedSetRangeStore), r=> r.SortedSetRangeStore);
+
+            var db = conn.GetDatabase();
+            var me = Me();
+            var sourceKey = $"{me}:ZSetSource";
+            var destinationKey = $"{me}:ZSetDestination";
+
+            db.KeyDelete(new RedisKey[] {sourceKey, destinationKey}, CommandFlags.FireAndForget);
+            db.SortedSetAdd(sourceKey, lexEntries, CommandFlags.FireAndForget);
+            var res = db.SortedSetRangeAndStore(destinationKey, sourceKey, "a", "j", SortedSetOrder.ByLex, Exclude.Both);
+            var range = db.SortedSetRangeByRankWithScores(destinationKey);
+            Assert.Equal(8, res);
+            for (var i = 1; i <lexEntries.Length-1; i++)
+            {
+                Assert.Equal(lexEntries[i], range[i-1]);
+            }
+        }
+
+        [Fact]
+        public void SortedSetRangeStoreByLexRevRange()
+        {
+            using var conn = Create();
+            Skip.IfMissingFeature(conn, nameof(RedisFeatures.SortedSetRangeStore), r=> r.SortedSetRangeStore);
+
+            var db = conn.GetDatabase();
+            var me = Me();
+            var sourceKey = $"{me}:ZSetSource";
+            var destinationKey = $"{me}:ZSetDestination";
+
+            db.KeyDelete(new RedisKey[] {sourceKey, destinationKey}, CommandFlags.FireAndForget);
+            db.SortedSetAdd(sourceKey, lexEntries, CommandFlags.FireAndForget);
+            var res = db.SortedSetRangeAndStore(destinationKey, sourceKey, "j", "a", SortedSetOrder.ByLex, Exclude.None, Order.Descending);
+            var range = db.SortedSetRangeByRankWithScores(destinationKey);
+            Assert.Equal(10, res);
+            for (var i = 0; i < lexEntries.Length; i++)
+            {
+                Assert.Equal(lexEntries[i], range[i]);
+            }
+        }
+
+        [Fact]
+        public void SortedSetRangeStoreFailErroneousTake()
+        {
+            using var conn = Create();
+            Skip.IfMissingFeature(conn, nameof(RedisFeatures.SortedSetRangeStore), r=> r.SortedSetRangeStore);
+
+            var db = conn.GetDatabase();
+            var me = Me();
+            var sourceKey = $"{me}:ZSetSource";
+            var destinationKey = $"{me}:ZSetDestination";
+
+            db.KeyDelete(new RedisKey[] {sourceKey, destinationKey}, CommandFlags.FireAndForget);
+            db.SortedSetAdd(sourceKey, lexEntries, CommandFlags.FireAndForget);
+            var exception = Assert.Throws<ArgumentException>(()=>db.SortedSetRangeAndStore(destinationKey, sourceKey,0,-1, take:5));
+            Assert.Equal($"take argument is not valid when sortedSetOrder is ByRank you may want to try setting the SortedSetOrder to ByLex or ByScore{Environment.NewLine}Parameter name: take", exception.Message);
```

Comment:
This may be built on one OS and run on another - if new lines are in tests you'll want to use the optional param:
```cs
Assert.Equal("", "", ignoreLineEndingDifferences: true);
``

Reply:
I figured out what it was, it's not an OS thing, it's a Framework vs core thing lol. framework's Argument 

good ol' framework uses environment newline: https://github.com/microsoft/referencesource/blob/master/mscorlib/system/argumentexception.cs#L70

whereas core runtime uses a space:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/ArgumentException.cs#L86

lol

Reply:
I just said, use the param name, good enough lol.